### PR TITLE
Refactor/answering machine

### PR DESCRIPTION
**Pull request by influx6**
PR updates the package to the new event API being used.

### CODE REVIEW

File: proxies/ssh/ssh_recorder.go [hunk 1]
```diff
@@ -68,7 +72,7 @@ func (r *SSHRecorder) NewSession(c *proxies.ProxyConn) *SSHRecorderSession {
 // Connect records the connect operation for the underline ssh connection.
 func (rs *SSHRecorderSession) Connect() {
 	rs.r.Channel.Send(
-		message.EventCategoryType(
+		event.Apply(
```
Comment (nl5887) [inline proxies/ssh/ssh_recorder.go]: How is event.Apply different then event.New?

File: server/ping.go [hunk 2]
```diff
@@ -3,15 +3,15 @@ package server
...
 )
 
 // ping delivers a ping event to the server indicate it's alive.
 func (hc *Honeytrap) ping() error {
-	hc.events.Send(message.BasicEvent{
-		Sensor: message.PingSensor,
-		Type:   message.PingEvent,
-	})
+	hc.events.Send(event.New(
+		event.Sensor(event.PingSensor),
```
Comment (nl5887) [inline server/ping.go]: Maybe event.PingSensor can be the sensor already? 

```
PingSensor = event.Sensor("ping")
```

File: server/ping.go [hunk 3]
```diff
@@ -3,15 +3,15 @@ package server
...
+		event.Type(event.PingEvent),
```
Comment (nl5887) [inline server/ping.go]:
```
PingEvent = event.Type("ping")
```

File: server/honeytrap.go [hunk 4]
```diff
@@ -141,25 +141,27 @@ func (hc *Honeytrap) startProxies() {
...
 
-			hc.events.Send(message.BasicEvent{
-				Sensor: message.EventSensor(st.Service),
-				Type:   message.ServiceStarted,
-				Details: map[string]interface{}{
+			hc.events.Send(event.New(
+				event.Sensor(event.ServiceSensor),
+				event.Category(st.Service),
+				event.Type(event.ServiceStarted),
+				event.CopyFrom(map[string]interface{}{
```
Comment (nl5887) [inline server/honeytrap.go]: We don't need primitive here.

File: server/honeytrap.go [hunk 5]
```diff
@@ -141,25 +141,27 @@ func (hc *Honeytrap) startProxies() {
...
-					Sensor: message.EventSensor(st.Service),
-					Type:   message.ServiceStarted,
-					Details: map[string]interface{}{
+				hc.events.Send(event.New(
+					event.Sensor(event.ServiceSensor),
+					event.Category(st.Service),
+					event.Type(event.ServiceStarted),
+					event.CopyFrom(map[string]interface{}{
 						"primitive": primitive,
 						"error":     err.Error(),
```
Comment (nl5887) [inline server/honeytrap.go]: Why not use `event.Error(err)`?

File: pushers/backends/slack/slack_test.go [hunk 6]
```diff
@@ -21,29 +21,20 @@ const (
...
-	}
-
-	crum = message.BasicEvent{
-		Sensor:      "Crum Stream",
-		Category:    "WebRTC Crum Stream",
-		SessionID:   "4334334-3433434-34343-FUD",
-		ContainerID: "56454-5454UDF-2232UI-34FGHU",
-		Data:        "Hello World!",
-	}
+	blueChip = event.New(event.CopyFrom(map[string]interface{}{
```
Comment (nl5887) [inline pushers/backends/slack/slack_test.go]: Just use event.Sensor("BlueChip") and event.Category("Chip Integrated")

File: pushers/backends/slack/slack.go [hunk 7]
```diff
@@ -69,21 +70,26 @@ func NewWith(meta toml.MetaData, data toml.Primitive) (pushers.Channel, error) {
...
 
 	//Attempt to encode message body first and if failed, log and continue.
 	var messageBuffer bytes.Buffer
 
-	if data := mesg.DataReader(); data != nil {
-		io.Copy(&messageBuffer, data)
+	category := mesg["cateory"].(string)
+	sensor := mesg["sensor"].(string)
+	etype := mesg["type"].(string)
+	txmessage := mesg["message"].(string)
```
Comment (nl5887) [inline pushers/backends/slack/slack.go]:
```
txmessage = fmt.Sprintf("Event with Category %q of Type %q for Sensor %q occured", category, etype, sensor)

if s, ok := mesg["message"]; !ok {
} else if s, ok := s.(string); !ok {
} else {
newmesg.Text = s
}
```

File: pushers/backends/slack/slack.go [hunk 8]
```diff
@@ -69,21 +70,26 @@ func NewWith(meta toml.MetaData, data toml.Primitive) (pushers.Channel, error) {
 
 // Send delivers the giving push messages to the required slack channel.
 // TODO: Ask if Send shouldnt return an error to allow proper delivery validation.
-func (mc SlackBackend) Send(mesg message.Event) {
+func (mc SlackBackend) Send(mesg event.Event) {
```
Comment (nl5887) [inline pushers/backends/slack/slack.go]: mesg = evt, newmesg = msg

File: proxies/ssh/ssh_recorder.go (hunk 1 above)
Comment (influx6) [inline proxies/ssh/ssh_recorder.go]: New generates an event object directly with the options, why apply does not. It expects to be supplied the Event to be used and applies the option. Useful for cases where we already have a function returning an Event but want to add more details to without manually calling each function individually.

File: server/ping.go (hunk 2 above)
Comment (influx6) [inline server/ping.go]: Alright.

File: server/honeytrap.go (hunk 5 above)
Comment (influx6) [inline server/honeytrap.go]: Alright, do-able.

**Pull request closed without merging**